frontend/packages/kubevirt-plugin/src/components/modals/uninstall-kubevirt-modal/uninstall-kubevirt-modal.tsx

```diff
@@ -0,0 +1,167 @@
+  };
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+
+    if (errorMessage !== '') {
+      setShowErrorMessage(true);
+      return;
+    }
+
+    const promise = new Promise((resolve) => {
+      // TODO Add uninstallation call
+      console.log(event); // eslint-disable-line no-console
+      resolve();
+    });
+
+    handlePromise(promise).then(() => close()); // eslint-disable-line promise/catch-or-return
+  };
+
+  return (
+    <form onSubmit={onSubmit} name="form" className="modal-content co-catalog-install-modal">
+      <ModalTitle className="modal-header">
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> {title}
+      </ModalTitle>
+      <ModalBody>
+        <p>
+          By uninstalling this operator, all virtual machines will be deleted along with their
+          storage. This action cannot be reversed and will result in data loss. Virtualization will
+          no longer be available on this cluster.
+        </p>
+        {numOfVMSOnCluster ? (
+          <>
+            <p>
+              <ExternalLink
+                href="/k8s/all-namespaces/virtualization"
+                text={`${numOfVMSOnCluster} virtual machines found on this cluster`}
+              />
+            </p>
+            <p>
+              Confirm uninstallation and deletion of all virtual machines by typing{' '}
+              <strong>{CONFIRMATION_TEXT}</strong> below:
+            </p>
+            <FormGroup
+              className="pf-m-3-col-on-md"
+              fieldId="confirm-vm-deletion-text-input"
+              id="confirm-vm-deletion-text-input"
+              isValid={!!errorMessage}
+              validated="error"
+            >
+              <TextInput
+                type="text"
+                value={confirmationTextInput}
+                onChange={(v) => onConfirmTextInputChange(v)}
+                aria-label="confirmation input"
+              />
+            </FormGroup>
+          </>
+        ) : (
+          <strong>No virtual machines found on this cluster</strong>
+        )}
+      </ModalBody>
+      <ModalSubmitFooter
+        inProgress={inProgress}
+        errorMessage={showErrorMessage ? errorMessage : null}
+        cancel={cancel}
+        submitDanger
+        submitText="Uninstall"
+      />
+    </form>
+  );
+});
+
+const resources = [
+  {
+    kind: VirtualMachineModel.kind,
+    prop: 'vms',
+    isList: true,
+  },
+  {
+    kind: VirtualMachineInstanceModel.kind,
+    prop: 'vmis',
+    isList: true,
+  },
+];
+
+const UninstallKubevirtModalWrapped = (props) => (
+  <Firehose resources={resources}>
+    <UninstallKubevirtModal {...props} />
+  </Firehose>
+);
+
+export type UninstallKubevirtModalProps = HandlePromiseProps &
+  ModalComponentProps & {
+    title?: string;
+    vms?: FirehoseResult<VMKind[]>;
+    vmis?: FirehoseResult<VMIKind[]>;
+  };
+
+const UninstallKubevirtModalConnected = connect()(UninstallKubevirtModalWrapped);
```

Comment:
I think It should work without the `connect()` just:
``` js
export const UninstallKubevirtModalEnhanced = createModalLauncher(UninstallKubevirtModalWrapped);
```
without this line, can you check ?